src/addrman.cpp

```diff
@@ -246,12 +246,17 @@ void AddrManImpl::Unserialize(Stream& s_)
 
     uint8_t compat;
     s >> compat;
+    if (compat < INCOMPATIBILITY_BASE) {
+        throw std::ios_base::failure(strprintf("Corrupted addrman database: The compat value (%u) "
+                                               "is lower than the expected %u.",
+                                               compat, INCOMPATIBILITY_BASE));
+    }
     const uint8_t lowest_compatible = compat - INCOMPATIBILITY_BASE;
     if (lowest_compatible > FILE_FORMAT) {
         throw std::ios_base::failure(strprintf(
             "Unsupported format of addrman database: %u. It is compatible with formats >=%u, "
             "but the maximum supported by this version of %s is %u.",
-            uint8_t{format}, uint8_t{lowest_compatible}, PACKAGE_NAME, uint8_t{FILE_FORMAT}));
+            format, lowest_compatible, PACKAGE_NAME, uint8_t{FILE_FORMAT}));
```

Comment:
The `format` variable is of type `enum Format : uint8_t` and I think it hit that bug in tinyformat. It printed a symbol with ASCII code 1 instead of the number 1.

Reply:
thanks, fixed